Core/HLE/proAdhoc.cpp

```diff
@@ -983,6 +988,54 @@ void freeFriendsRecursive(SceNetAdhocctlPeerInfo * node) {
 	free(node);
 }
 
+//@params chatmenu pass NULL on destroy, and pass ChatMenu On Create (EmuScreen.cpp)
+void setChatPointer(ChatMenu * chatmenu) {
+	if (chatmenu != NULL) {
```

Comment:
this pointer really give me a headache its only quick workaround to update the chat screen ui  from proadhoc.cpp when new chat send or new chat is coming that i found. didn't know how to properly linking class and file on C++ because im still a newcomer in C++ programming and PPSSPP development. this function also cause a random crash with access violation reading location or access violation writing location. so im using null as workaround on chatscreen destructor. but i think it cause memory leak and try that if argument delete but it still crashed on low end device (already publish a build to online community for helping me testing the code) most of them report 1 or 2 random crash occured in 1 day. i will try remove this function completely and implement your below comment it looks a good solution thanks for reviewing :) also i have some question below